src/init.cpp

```diff
@@ -1749,7 +1750,11 @@ bool AppInitMain(NodeContext& node, interfaces::BlockAndHeaderTipInfo* tip_info)
     connOptions.nReceiveFloodSize = 1000 * args.GetIntArg("-maxreceivebuffer", DEFAULT_MAXRECEIVEBUFFER);
     connOptions.m_added_nodes = args.GetArgs("-addnode");
 
-    connOptions.nMaxOutboundLimit = 1024 * 1024 * args.GetIntArg("-maxuploadtarget", DEFAULT_MAX_UPLOAD_TARGET);
+    auto opt_max_upload = ParseByteUnits(args.GetArg("-maxuploadtarget", DEFAULT_MAX_UPLOAD_TARGET));
+    if (!opt_max_upload) {
```

Comment:
I want to point out to reviewers that this changes start up behavior. Previously,  `-maxuploadtarget`  would silently fail and use DEFAULT_MAX_UPLOAD_TARGET. 
This change will throw an error upon parse failure, but only  after 30-50 seconds after start-up

Reply:
Moved